runtime/src/main/java/org/corfudb/runtime/view/stream/ThreadSafeStreamView.java

```diff
@@ -0,0 +1,106 @@
+package org.corfudb.runtime.view.stream;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.StreamOptions;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+
+/**
+ * A thread-safe implementation of IStreamView.
```

Comment:
Can we add a comment on which methods are made thread-safe with regards to BackpointerStreamView, it would ease comparison of both.

Reply:
BackpointerStreamView is hidden, the IStreamView interface is thread-safe (i.e. all methods).